kore/src/Kore/Step/Simplification/Overloading.hs

```diff
@@ -83,19 +86,37 @@ overloadedAndEqualsOverloading
     termMerger
     tools
     first@(App_ firstHead _)
-    second@(App_ sortInjection [App_ secondHead secondChildren])
+    (App_ sortInjection [App_ secondHead secondChildren])
   | MetadataTools.isOverloading tools firstHead secondHead
   = equalInjectiveHeadsAndEquals
         termMerger
         first
         (resolveOverloading sortInjection firstHead secondHead secondChildren)
-  | MetadataTools.isConstructorOrOverloaded tools secondHead
+overloadedAndEqualsOverloading
+    _termMerger
+    tools
+    first@(App_ _ _)
+    second@(App_ _ [child])
+  | Just typeName <- notUnifiableType child
   = Monad.Trans.lift $ do
         explainBottom
-            "Cannot unify overloaded constructor with different injected ctor."
+            ("Cannot unify overloaded constructor with " <> typeName <> ".")
             first
             second
         empty
+  where
+    notUnifiableType :: TermLike variable -> Maybe (Doc ())
+    notUnifiableType (App_ appHead _)
+      | MetadataTools.isConstructorOrOverloaded tools appHead
+      = Just "different injected ctor"
+    notUnifiableType (DV_ _ _) = Just "injected domain value"
+    notUnifiableType (BuiltinBool_ _) = Just "injected builtin bool"
+    notUnifiableType (BuiltinInt_ _) = Just "injected builtin int"
+    notUnifiableType (BuiltinList_ _) = Just "injected builtin list"
```

Comment:
If one overloads the list constructor, couldn't they unify against a builtin list?

Reply:
actually, I think I know the answer to the question above: builtin list is List{K} and cannot be overloaded, right?

Reply:
I was asking myself the same question, and I had roughly the same answer. But since we both doubted this, maybe it's safer to remove these and to add them only when needed.